projects/director/director_agent.py

```diff
@@ -26,19 +26,38 @@
 import parlai.utils.logging as logging
 
 
+class ScalarLayer(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.params = nn.Parameter(torch.Tensor([1.0, 0.0]))
+
+    def forward(self, input: torch.Tensor):
+        return input * self.params[0].expand_as(input) + self.params[1].expand_as(input)
+
+
 class DirectorModel(TransformerGeneratorModel):
     """
     Director model that extends TransformerGeneratorModel and adds |V| binary classifier
     heads.
     """
 
-    def __init__(self, opt: Opt, dictionary: DictionaryAgent, **kwargs):
+    def __init__(
+        self,
+        opt: Opt,
+        dictionary: DictionaryAgent,
+        use_shared_embedding=False,
+        **kwargs,
+    ):
         super().__init__(opt, dictionary, **kwargs)
 
         vocabulary_size = len(dictionary)
 
         decoder_output_dim = self.decoder.out_dim
-        self.classifier_heads = nn.Linear(decoder_output_dim, vocabulary_size)
+        self.use_shared_embedding = use_shared_embedding
```

Comment:
if `use_shared_embedding` is passing from opt, then you can make sure of that  `self.use_shared_embedding = opt.get('use_shared_embedding', False)?

Reply:
Thanks for the comment. I changed it.